---
title: "Look for genes with variants in multiple families"
author: "EF"
date: "29 Jan 2018"
output: html_document
---
# start_section
  
```{r start_section}

# Time stamp
Sys.time()

# Clean up
rm(list=ls())
graphics.off()

# Set root working folder
library(knitr)
library(dplyr)
library(stringr)
library(kableExtra)

options(scipen=999)
options(knitr.table.format = "html") 

opts_knit$set(root.dir = "/rds/project/erf33/rds-erf33-medgen/users/eleanor/2015-16/IHCAP/IHCAP_sets/IHCAP_ACC_Jan18/")
setwd("/rds/project/erf33/rds-erf33-medgen/users/eleanor/2015-16/IHCAP/IHCAP_sets/IHCAP_ACC_Jan18/")

```

#Load image
```{r load image}
#Load data
load("R_data/s03_ihcap_acc_Jan18_oncocytic_v2_20180116.Rdata")

vv <- aggr_acc
gt <- gt.onco

#Remove unwanted data
rm(aggr, merged, aggr_acc, gt.onco)

#Check dimensions
dim(gt)
dim(vv)

#Check rows
sum(row.names(gt) != row.names(vv)) 

#Number of genes
length(unique(vv$SYMBOL))

#Load 1000 genomes data
kgen.df <- read.table("/rds/project/erf33/rds-erf33-medgen/users/eleanor/2015-16/1kgenomes_analysis/1000genomes_variantfilter_EUR_20170330.csv", sep = ",", header = TRUE, stringsAsFactors = FALSE)

#Add VperG column
kgen <- kgen.df %>% group_by(SYMBOL) %>% mutate(VperG = length(SYMBOL))

rm(kgen.df)

```

#Create aggregated allele frequencies in kgen
```{r kgen af}
#Number of rare protein affecting variants in kgen
nrow(kgen)

#Number genes in kgen
length(unique(kgen$SYMBOL))
       
#Create vperg table of each gene with one high impact variant
kgen.vpg <- subset(kgen, select=c("SYMBOL", "VperG")) %>% unique(.)

#Find top 1% VpG scores
m <- round(quantile(kgen.vpg$VperG, 0.99),digits=0)

#Select variants not in top 1% variable and < 0.01 in eurAF
kgen_v <- kgen[kgen$VperG  < m & kgen$EUR_AF <= 0.01,]

#Number of variants not in top 1% variable
nrow(kgen_v)

#Number of genes not in top 1% variable
length(unique(kgen_v$SYMBOL))

#Select gene names and AFs
kgen_short <- subset(kgen_v, select=c("SYMBOL", "EUR_AF"))

#calculate ACs from AF and AN
kgen_short$AN <- 503*2 #(503 Europeans in kgen phase 3)
kgen_short$kgen_AC <- round(kgen_short$EUR_AF*kgen_short$AN)

#Aggregate AC per gene
aggr_kgen <- kgen_short %>% group_by(SYMBOL) %>% mutate(kgen_AC=sum(kgen_AC, na.rm=TRUE))

#Calculate without variant
aggr_kgen$kgen_without_AC <- aggr_kgen$AN - aggr_kgen$kgen_AC

#Create list of all genes with counts
kgen.c <- unique(select(aggr_kgen, -EUR_AF, -AN))

#Write file for input into GM cytoscape
write.csv(kgen.c, "kgen_eur_genes_cytoscape_rare_v2_20180407.csv")

rm(kgen, kgen_short, aggr_kgen, kgen.vpg, m, kgen_v)

```

#Create aggregated allele frequences in acc data
```{r acc af}

#Create vperg table of each gene with one high impact variant
gene.vpg <- subset(vv, select=c("SYMBOL", "VperG")) %>% unique(.)

#Find top 1% VpG scores
n <- round(quantile(gene.vpg$VperG, 0.99),digits=0)

#Create set of variants after removal of top 1% (plus AC less than 5 in whole ihcap set) 
vv_v <- vv[vv$VperG  < n & as.numeric(as.character(vv$AC)) < 5 & (vv$kgen.EUR_AF <= 0.01 | is.na(vv$kgen.EUR_AF)),]

#Number of variants in this set
nrow(vv_v)

#Number of genes in this set
length(unique(vv_v$SYMBOL))

#Select short subset
vv_short <- subset(vv_v, select=c("SYMBOL", "ACC_AC", "ACC_AN"))

#Aggregate allele counts into genes
aggr_vv <- vv_short %>% group_by(SYMBOL) %>% mutate(ACC_AC=sum(ACC_AC, na.rm=TRUE), ACC_AN=round(mean(ACC_AN, na.rm=TRUE)))

#Create count without variant
aggr_vv$ACC_without_AC <- aggr_vv$ACC_AN - aggr_vv$ACC_AC

#Tidy
vv.c <- unique(select(aggr_vv, -ACC_AN))

#Write file for input into GM cytoscape
write.csv(vv.c, "ACC_genes_cytoscape_rare_v2_20180129.csv")

rm(vv, aggr_vv, vv_short, gene.vpg, n)
```


#merge ACs from both sets
```{r merge}
#Merge allele counts from both sets
all.c <- as.matrix(merge(vv.c, kgen.c, by="SYMBOL", all = TRUE))

#Set missing ACs to 0
all.c[is.na(all.c)] <- 0
all.df <- as.data.frame(all.c)

#convert all factors back to numerics
all.df$ACC_AC <- as.numeric(as.character(all.df$ACC_AC))
all.df$ACC_without_AC <- as.numeric(as.character(all.df$ACC_without_AC))

all.df$kgen_AC <- as.numeric(as.character(all.df$kgen_AC))
all.df$kgen_without_AC <- as.numeric(as.character(all.df$kgen_without_AC))

#if kgen ac is 0, set without to 1006
all.df$kgen_without_AC[all.df$kgen_AC == 0] <- 1006
all.df$kgen_without_AC[all.df$kgen_without_AC < 0] <- 0
all.df$ACC_without_AC[all.df$ACC_AC == 0] <- 16
all.df$ACC_without_AC[all.df$ACC_without_AC < 0] <- 0

all.df <- subset(all.df, select=c("SYMBOL", "ACC_AC", "kgen_AC", "ACC_without_AC", "kgen_without_AC"))

rm(all.c, kgen.c, vv.c)
```

#Load list of DNAR genes
```{r dnar}
#Load gene lists
assem <- read.table("GO_lists/celljunc_assem_go0034329_genes.txt", sep = "\t", header = TRUE, stringsAsFactors = FALSE)
mapk <- read.table("GO_lists/mapk_go0000187_genes.txt", sep = "\t", header = TRUE, stringsAsFactors = FALSE)
mirna <- read.table("GO_lists/miRNA_silencing_go0060964_gene.txt", sep = "\t", header = TRUE, stringsAsFactors = FALSE)
muscle <- read.table("GO_lists/muscle_dev_go0055001_genes.txt", sep = "\t", header = TRUE, stringsAsFactors = FALSE)


#Create files of each GO term
assem.ac <- filter(all.df, SYMBOL %in% assem$Gene)
mapk.ac <- filter(all.df, SYMBOL %in% mapk$Gene)
mirna.ac <- filter(all.df, SYMBOL %in% mirna$Gene)
muscle.ac <- filter(all.df, SYMBOL %in% muscle$Gene)

#Number of total genes with variants in each set
nrow(assem.ac)
nrow(mapk.ac)
nrow(mirna.ac)
nrow(muscle.ac)

```

#Create dataframe with pathways
```{r pathways}
#Create one DF for GO counts
paths <- data.frame(Term=c("ASSEM", "MAPK", "miRNA", "MUSCLE"), ACC_AC=c(sum(assem.ac$ACC_AC), sum(mapk.ac$ACC_AC), sum(mirna.ac$ACC_AC), sum(muscle.ac$ACC_AC)), kgen_AC=c(sum(assem.ac$kgen_AC), sum(mapk.ac$kgen_AC), sum(mirna.ac$kgen_AC), sum(muscle.ac$kgen_AC)))

#Reset without ACs
paths$ACC_without_AC <- 16 - paths$ACC_AC
paths$kgen_without_AC <- 1006 - paths$kgen_AC
```

#Fishers exact
```{r FE}
#Calculate pvalues using two and one sided fishers
paths$pval <- paths %>% apply(1, function(x){
  fishtable <- matrix(as.numeric(c(x[2:5])), nrow = 2, byrow=TRUE)
  fisher.test(fishtable)[[1]]
})

paths$pval_onesided <- paths %>% apply(1, function(x){
  fishtable <- matrix(as.numeric(c(x[2:5])), nrow = 2, byrow=TRUE)
  fisher.test(fishtable, alternative = "greater")[[1]]
})

paths$FDR.p <- signif(p.adjust(paths$pval, method="fdr", n=nrow(paths)), digits=2)

paths

kable(paths, col.names=c("GO Term", "ACC with variant", "Control with variant","ACC without variant", "Control without variant", "P value", "One-sided P value", "FDR P value")) %>% column_spec(1, bold = T) %>%  kable_styling(bootstrap_options = "striped", full_width = F)
```

